ITunesSearch/ITunesSearch/SearchResultController.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  SearchResultController.swift
+//  ITunesSearch
+//
+//  Created by Taylor Lyles on 5/14/19.
+//  Copyright © 2019 Taylor Lyles. All rights reserved.
+//
+
+import Foundation
+
+
+class SearchResultController {
+	let baseURL = URL(string: "https://itunes.apple.com/search")!
+	var searchResults: [SearchResult] = []
+	
+	func performSearch(searchTerm: String, resultType: ResultType, completion: @escaping (Error?) -> Void) {
+		
+		let searchURL = baseURL.appendingPathComponent("searchResults")
+		
+		var urlComponents = URLComponents(url: searchURL, resolvingAgainstBaseURL: true)
+
+		let searchQueryItem = URLQueryItem(name: "term", value: searchTerm)
+		
+		let searchQueryType = URLQueryItem(name: "entity", value: resultType.rawValue)
+		urlComponents?.queryItems = [searchQueryItem, searchQueryType]
+		
+
+		guard let formattedURL = urlComponents?.url else {
+			NSLog("Not getting url")
+			completion(nil)
```

Comment:
We don't want to pass nil here because then where we call this will assume there is nothing as in don't do anything. instead we want to create an error, you can do that with NSError()

`completion(NSError())`